alerting/src/main/kotlin/org/opensearch/alerting/DocumentLevelMonitorRunner.kt

```diff
@@ -210,17 +210,27 @@ object DocumentLevelMonitorRunner : MonitorRunner() {
         val idQueryMap: Map<String, DocLevelQuery> = queries.associateBy { it.id }
 
         val triggerResults = mutableMapOf<String, DocumentLevelTriggerRunResult>()
-        monitor.triggers.forEach {
-            triggerResults[it.id] = runForEachDocTrigger(
-                monitorCtx,
-                monitorResult,
-                it as DocumentLevelTrigger,
-                monitor,
-                idQueryMap,
-                docsToQueries,
-                queryToDocIds,
-                dryrun
-            )
+        // If there are no triggers defined, we still want to generate findings
+        if (monitor.triggers.isEmpty()) {
+            if (dryrun == false && monitor.id != Monitor.NO_ID) {
+                docsToQueries.forEach {
+                    val triggeredQueries = it.value.map { queryId -> idQueryMap[queryId]!! }
+                    createFindings(monitor, monitorCtx, triggeredQueries, it.key, true)
```

Comment:
why are we even creating findings inside this method?
If we are trying to decouple findings from triggers, why not create findings when they are available?
Do we want to ensure that we don't create findings if triggers fail or something?

Reply:
If we are not refactoring this now, can you link the github issue to follow up on this in the reply to this comment?

Reply:
#865